e2e-polybft/e2e/bridge_test.go

```diff
@@ -287,14 +285,164 @@ func TestE2E_Bridge_Transfers(t *testing.T) {
+		require.NoError(t, err)
+
+		// Deploy ERC20 contract to the rootchain
+		rootchainDeployer, err := rootHelper.DecodePrivateKey("")
+		require.NoError(t, err)
+
+		txn := &ethgo.Transaction{To: nil, Input: contractsapi.RootERC20.Bytecode}
+		receipt, err := rootchainTxRelayer.SendTransaction(txn, rootchainDeployer)
+		require.NoError(t, err)
+		require.NotNil(t, receipt)
+
+		contractAddress := receipt.ContractAddress
+
+		t.Log("Rootchain token address:", contractAddress)
+
+		// wait for next sprint block as the starting point,
+		// in order to be able to make assertions against blocks offsetted by sprints
+		initialBlockNum, err := childEthEndpoint.BlockNumber()
+		require.NoError(t, err)
+
+		initialBlockNum = initialBlockNum + sprintSize - (initialBlockNum % sprintSize)
+		require.NoError(t, cluster.WaitForBlock(initialBlockNum, 1*time.Minute))
+
+		t.Log("Initial block number:", initialBlockNum)
+
+		// get initial exit id
+		initialExitEventID := getLastExitEventID(t, childchainTxRelayer)
+
+		// send a few transactions to the bridge
+		require.NoError(t, cluster.Bridge.Deposit(
+			common.ERC20,
+			types.Address(contractAddress),
+			polybftCfg.Bridge.RootERC20PredicateAddr,
+			rootHelper.TestAccountPrivKey,
+			strings.Join(receivers[:], ","),
+			strings.Join(amounts[:], ","),
+			"",
+			cluster.Bridge.JSONRPCAddr(),
+			rootHelper.TestAccountPrivKey,
+			false,
+		))
+
+		// wait for a few more sprints
+		finalBlockNumber := initialBlockNum + 10*sprintSize
+		require.NoError(t, cluster.WaitForBlock(finalBlockNumber, 2*time.Minute))
+
+		t.Log("Final block number:", finalBlockNumber)
+
+		// the transactions are processed and there should be a success events
+		logs, err := getFilteredLogs(stateSyncedResult.Sig(), initialBlockNum, finalBlockNumber, childEthEndpoint)
+		require.NoError(t, err)
+
+		// assert that all deposits are executed successfully
+		// deposit transaction only
+		checkStateSyncResultLogs(t, logs, transfersCount+1)
+
+		// retrieve child token address
+		rootToChildTokenFn := contractsapi.ChildERC20Predicate.Abi.Methods["rootTokenToChildToken"]
+		input, err := rootToChildTokenFn.Encode([]interface{}{contractAddress})
+		require.NoError(t, err)
+
+		childTokenRaw, err := childchainTxRelayer.Call(ethgo.ZeroAddress, ethgo.Address(contracts.ChildERC20PredicateContract), input)
+		require.NoError(t, err)
+
+		childTokenAddr := types.StringToAddress(childTokenRaw)
+
+		t.Log("Childchain token address:", childTokenAddr)
+
+		// check receivers balances got increased by deposited amount
+		for _, receiver := range receiversAddrs {
+			balanceOfFn := &contractsapi.BalanceOfChildERC20Fn{Account: receiver}
+
+			balanceInput, err := balanceOfFn.EncodeAbi()
+			require.NoError(t, err)
+
+			balanceRaw, err := childchainTxRelayer.Call(ethgo.ZeroAddress, ethgo.Address(childTokenAddr), balanceInput)
+			require.NoError(t, err)
+
+			balance, err := types.ParseUint256orHex(&balanceRaw)
+			require.NoError(t, err)
```

Comment:
This part can be simplified:
```suggestion
	balance := erc20BalanceOf(t, receiver, childTokenAddr, childchainTxRelayer)
```

Reply:
Done